tasks/install-provider-rhv.yml

```diff
@@ -1,12 +1,23 @@
 ---
 - name: Look for SSH key
   stat:
-    path: "~vdsm/.ssh/id_rsa"
+    path: "/var/lib/vdsm/.ssh/id_rsa"
   ignore_errors: "yes"
   register: stat_ssh_key
 
+- name: Create .ssh directory
+  file:
+    path: /var/lib/vdsm/.ssh
+    owner: vdsm
+    group: kvm
+    mode: 0700
+    state: present
+  become: true
+  become_user: vdsm
```

Comment:
Not sure why you use `become`. The playbook is supposed to be run as `root`, and you already set the owner and group.